### ISSUE TITLE
tcl-callback.t failure reports

### ISSUE THREAD

**Issue by chrstphrchvz** [chrstphrchvz/tkx#1]
http://www.cpantesters.org/cpan/report/efb5ffb8-6892-11eb-8ad4-32611f24ea8f
http://www.cpantesters.org/cpan/report/0bb12ada-6893-11eb-8ad4-32611f24ea8f
http://www.cpantesters.org/cpan/report/fa9ea3da-6892-11eb-8ad4-32611f24ea8f
http://www.cpantesters.org/cpan/report/061f5236-6893-11eb-8ad4-32611f24ea8f
http://www.cpantesters.org/cpan/report/0091a04e-6893-11eb-8ad4-32611f24ea8f
http://www.cpantesters.org/cpan/report/f529aa80-6892-11eb-8ad4-32611f24ea8f
http://www.cpantesters.org/cpan/report/297f8a1e-686d-11eb-9df2-f6591f24ea8f
http://www.cpantesters.org/cpan/report/11323f9e-6893-11eb-8ad4-32611f24ea8f
http://www.cpantesters.org/cpan/report/24a08206-691b-11eb-a1db-b18c1f24ea8f
http://www.cpantesters.org/cpan/report/29c44ef2-691b-11eb-a1db-b18c1f24ea8f
http://www.cpantesters.org/cpan/report/20581ae2-691b-11eb-a1db-b18c1f24ea8f

```
Tcl error 'invalid command name "::perl::CODE(0x801ca8f90)" at /usr/perl5.18.4p/lib/site_perl/5.18.4/amd64-freebsd/Tcl.pm line 765.
' while invoking scalar result call:
	"eval [set foo] a {b c}" at t/tcl-callback.t line 14.
t/tcl-callback.t .. 
Dubious, test returned 25 (wstat 6400, 0x1900)
Failed 6/7 subtests 
```

https://github.com/chrstphrchvz/tkx/blob/2a2dfd1182249a179a8de8d28124fe0a7effa5f6/t/tcl-callback.t#L9-L14

Not sure what could be causing this. Tester appears to have observed this on FreeBSD 10.1 using Perl 5.18.4/5.20.1/5.20.2/5.20.3/5.22.0/5.22.1/5.22.2/5.24.0/5.24.1; FreeBSD 10.3 using Perl 5.24.4/5.26.0; all using Tcl.pm 1.06; not sure which Tcl/Tk version(s).

Anyone observing this on a supported platform? (FreeBSD 10.x is unsupported for some time.)

Is it observed with a newer Tcl.pm? (Is this test case better off moved to Tcl.pm?)

**Comment by chrstphrchvz** [comment]
@eserte, could you please check on the affected testers whether this error always happens or just some of the time, and whether updating Tcl.pm resolves it?

**Comment by eserte** [comment]
The package tk84-8.4.20,2 is installed here.

On this machine there were also six reports where Tcl 1.27 was used. Here the test suite fails in a different way:
```
Tcl error 'version conflict for package "Tcl": have 8.6.6, need exactly 8.4 at /home/cpansand/.cpan/build/2021020622/Tcl-1.27-mTkETK/blib/lib/Tcl.pm line 817.
' while invoking scalar result call:
        "package require Tk" at t/LabEntry.t line 7.
Compilation failed in require at t/LabEntry.t line 7.
BEGIN failed--compilation aborted at t/LabEntry.t line 7.
t/LabEntry.t ...... 
Dubious, test returned 2 (wstat 512, 0x200)
No subtests run 
```
On this machine two tcl packages seem to be installed. "pkg info" lists tcl84-8.4.20_2,1 and tcl86-8.6.6_2.

**Comment by chrstphrchvz** [comment]
I would consider the package version conflict issue to be part of #2.

**Comment by chrstphrchvz** [comment]
I released Tkx 1.10_02 with tcl-callback0.t, a Tcl.pm-only equivalent of tcl-callback.t. Example failure report: http://www.cpantesters.org/cpan/report/b1418886-6dda-11eb-921d-87891f24ea8f

```
Tcl error 'invalid command name "::perl::CODE(0x801cc31e0)" at /usr/perl5.20.3p/lib/site_perl/5.20.3/amd64-freebsd/Tcl.pm line 765.
' while invoking scalar result call:
	"eval [set foo] a {b c}" at t/tcl-callback.t line 14.
t/tcl-callback.t ... 
Dubious, test returned 25 (wstat 6400, 0x1900)
Failed 6/7 subtests 
invalid command name "::perl::CODE(0x801ddcfa8)" at t/tcl-callback0.t line 19.
t/tcl-callback0.t .. 
Dubious, test returned 38 (wstat 9728, 0x2600)
Failed 6/7 subtests 
```

Tkx is not causing this issue. Maybe the issue is in Tcl.pm or Tcl, but I am not interested in reporting it elsewhere unless it is observed on a supported platform. I will consider requiring newer Tcl.pm (after improving Tkx's compatibility with it), and maybe that will avoid this issue.

**Comment by chrstphrchvz** [comment]
I believe this issue was the one discussed at length in https://rt.cpan.org/Ticket/Display.html?id=125472 and seemingly fixed in newer Tcl.pm (along with its t/set-callback.t based on Tkx's t/tcl-callback.t).
